activation/atxdb.go

```diff
@@ -192,8 +201,28 @@ func (db *ActivationDb) CalcActiveSetFromView(view []types.BlockID, pubEpoch typ
 	if found {
 		return count, nil
 	}
+	//check if we have a running calculation for this hash
+	db.assLock.Lock()
+	mu, alreadyRunning := db.pendingActiveSet[viewHash]
+	if alreadyRunning {
+		db.assLock.Unlock()
+		// if there is a running calculation, wait for it to end and get the result
+		mu.Lock()
+		count, found := activesetCache.Get(viewHash)
+		if found {
+			mu.Unlock()
+			return count, nil
+		}
+		// if not found, keep running mutex and calculate active set size
+	} else {
```

Comment:
If not running does not necessarily mean we don't have the calculation in the cache.

Reply:
this is checked at the beginning of the function